lib/scanner/walk.go

```diff
@@ -476,8 +479,11 @@ func SymlinkTypeEqual(disk, index uint32) bool {
 	// know means you are on Unix, and on Unix you don't really care what the
 	// target type is. The moment you do know, and if something doesn't match,
 	// that will propagate through the cluster.
-	if disk&protocol.FlagSymlinkMissingTarget != 0 && index&protocol.FlagSymlinkMissingTarget == 0 {
+	if disk == symlinks.TargetUnknown {
 		return true
 	}
-	return disk&protocol.SymlinkTypeMask == index&protocol.SymlinkTypeMask
+	if f.IsDirectory() {
+		return disk == symlinks.TargetDirectory
+	}
+	return disk == symlinks.TargetFile
```

Comment:
Ah, I see. But how did we handle this before? It seems we have two different cases for the same comparison:

Local Unknown, remote Known => These are equal, if we're scanning, because we don't want to override them with less data.

Local Unknown, remote Known => These should be unequal if we're pulling, as we should update our symlink on disk.